File: app/src/main/java/com/tasomaniac/openwith/data/Analytics.java
package com.tasomaniac.openwith.data;

import timber.log.Timber;

public interface Analytics {

    void sendScreenView(String screenName);

    void sendEvent(String category, String action, String label);

    class DebugAnalytics implements Analytics {

        @Override
        public void sendScreenView(String screenName) {
            Timber.tag("Analytics").d("Screen: %s", screenName);
        }

        @Override
        public void sendEvent(String category, String action, String label) {
            Timber.tag("Analytics").d("Event recorded:\n\tCategory: %s\n\tAction: %s\n\tLabel: %s", category, action, label);
        }
    }
}

File: app/src/main/java/com/tasomaniac/openwith/data/Analytics.kt
package com.tasomaniac.openwith.data

import timber.log.Timber

interface Analytics {

    fun sendScreenView(screenName: String)

    fun sendEvent(category: String, action: String, label: String)

    class DebugAnalytics : Analytics {

        override fun sendScreenView(screenName: String) {
            Timber.tag("Analytics").d("Screen: %s", screenName)
        }

        override fun sendEvent(category: String, action: String, label: String) {
            Timber.tag("Analytics").d(
                """\
                    |Event recorded:
                    |    Category: $category
                    |    Action: $action
                    |    Label: $label
                """.trimMargin()
            )
        }
    }
}
